apps/studio/src/components/ResourceSelector/useResourceQuery.tsx

```diff
@@ -0,0 +1,55 @@
+import { type ResourceItemContent } from "~/schemas/resource"
+import { trpc } from "~/utils/trpc"
+
+export const useResourceQuery = ({
+  siteId,
+  moveDest,
+  parentDest,
+  isResourceHighlighted,
+  onlyShowFolders,
+  resourceIds,
+}: {
+  siteId: number
+  moveDest: ResourceItemContent | undefined
+  parentDest: ResourceItemContent | undefined
+  isResourceHighlighted: boolean
+  onlyShowFolders: boolean
+  resourceIds?: ResourceItemContent["id"][]
+}) => {
+  const queryFn = onlyShowFolders
+    ? trpc.resource.getFolderChildrenOf.useInfiniteQuery
+    : trpc.resource.getChildrenOf.useInfiniteQuery
+  const {
+    data: { pages } = { pages: [{ items: [], nextOffset: null }] },
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage,
+  } = queryFn(
+    {
+      resourceId:
+        (isResourceHighlighted ? parentDest?.id : moveDest?.id) ?? null,
+      siteId: String(siteId),
+      limit: 25,
+    },
+    {
+      getNextPageParam: (lastPage) => lastPage.nextOffset,
+    },
+  )
+
+  const useResourceIdsFromSearch = !!resourceIds
+  const [resourceItemsWithAncestryStack] =
+    trpc.resource.getBatchAncestryWithSelf.useSuspenseQuery({
+      siteId: String(siteId),
+      resourceIds: useResourceIdsFromSearch
+        ? resourceIds
+        : pages.flatMap(({ items }) => items).map((item) => item.id),
+    })
+
+  return {
+    resourceItemsWithAncestryStack,
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    fetchNextPage: useResourceIdsFromSearch ? () => {} : fetchNextPage,
+    hasNextPage: useResourceIdsFromSearch ? false : hasNextPage,
+    isFetchingNextPage: useResourceIdsFromSearch ? false : isFetchingNextPage,
```

Comment:
why would we ever make this determination? the caller gets these functions/props back and is in charge of using them as-is. 

the caller also knows if we should `useResourceIdsFromSearch` as it's derived from something they pass (`!!resourceIds`) 

all of this implies that we should just eliminate the ternaries here and let the caller make this judgement themselves, no?